Update usage to print optional `--`

o-lim opened the pull request:
Update usage to print `[--]` signifying that `--` is an optional marker, indicating the end of options and the beginning of positional arguments.

amireh commented on src/cliargs.lua:
```diff
@@ -520,11 +520,14 @@ function cli:print_usage(noprint)
-    msg = msg .. " [OPTIONS] "
+    msg = msg .. " [OPTIONS]"
+  end
+  if self.required[1] or self.optargument.maxcount > 0 then
```
just curious, is there a reason we're not checking `#self.required` instead of `self.required[1]` ?

o-lim commented:
I just copied the `if self.required[1] then` below it, wasn't sure what the style was.  Just rebased to use `#self.required > 0` instead.  I also updated the condition proceeding it and other cases of `if self.optional[1] then` to `if #self.optional > 0 then`.
